tests/orchestrations_test.go

```diff
@@ -715,6 +715,142 @@ func Test_TerminateOrchestration(t *testing.T) {
 	)
 }
 
+func Test_TerminateOrchestration_Recursive(t *testing.T) {
+	delayTime := 4 * time.Second
+	r := task.NewTaskRegistry()
+	r.AddOrchestratorN("Root", func(ctx *task.OrchestrationContext) (any, error) {
+		ctx.CallSubOrchestrator("L1", task.WithSubOrchestrationInstanceID(string(ctx.ID)+"_L1")).Await(nil)
+		return nil, nil
+	})
+	r.AddOrchestratorN("L1", func(ctx *task.OrchestrationContext) (any, error) {
+		ctx.CallSubOrchestrator("L2", task.WithSubOrchestrationInstanceID(string(ctx.ID)+"_L2")).Await(nil)
+		return nil, nil
+	})
+	r.AddOrchestratorN("L2", func(ctx *task.OrchestrationContext) (any, error) {
+		ctx.CreateTimer(delayTime).Await(nil)
+		return nil, nil
+	})
+
+	// Initialization
+	ctx := context.Background()
+	client, worker := initTaskHubWorker(ctx, r)
+	defer worker.Shutdown(ctx)
+
+	// Test terminating with and without recursion
+	for _, recurse := range []bool{false} {
+		t.Run(fmt.Sprintf("Recurse = %v", recurse), func(t *testing.T) {
+			// Run the orchestration, which will block waiting for external events
+			id, err := client.ScheduleNewOrchestration(ctx, "Root")
+			require.NoError(t, err)
+
+			// Wait long enough to ensure all orchestrations have started (but not longer than the timer delay)
+			time.Sleep(3 * time.Second)
```

Comment:
Please, no `time.Sleep` to coordinate tests as this is going to be flaky. Instead, use something like `assert.Eventually` to poll until a condition is met

Reply:
Done